packages/vm/src/testing.rs

```diff
@@ -1,22 +1,26 @@
 //! This file has some helpers for integration tests.
 //! They should be imported via full path to ensure there is no confusion
 //! use cosmwasm_vm::testing::X
-use serde::Serialize;
-// JsonSchema is a flag for types meant to be publically exposed
 use schemars::JsonSchema;
+use serde::{de::DeserializeOwned, Serialize};
+use std::fmt;
 
 use cosmwasm_std::testing::{
     mock_dependencies, mock_dependencies_with_balances, MockApi, MockQuerier, MockStorage,
 };
 use cosmwasm_std::{
-    to_vec, Api, ApiError, Coin, Env, HandleResponse, HumanAddr, InitResponse, Querier,
+    to_vec, Api, ApiError, Coin, Env, HandleResponse, HumanAddr, InitResponse, NoMsg, Querier,
     QueryResponse, Storage,
 };
 
 use crate::calls::{call_handle, call_init, call_query};
 use crate::compatability::check_wasm;
 use crate::instance::Instance;
 
+// some helper types for parsing handle/init responses
+pub type HandleResult<U = NoMsg> = Result<HandleResponse<U>, ApiError>;
+pub type InitResult<U = NoMsg> = Result<InitResponse<U>, ApiError>;
```

Comment:
Arg, we already have those symbols in packages/std/src/init_handle.rs, but defined as

```
pub type InitResult = ApiResult<InitResponse>;
// ...
pub type HandleResult = ApiResult<HandleResponse>;
```

which is almost the same, just that ApiResult is not implemented via `Result`.

I'll have another look into `ApiResult` now.

Reply:
This looks promising. Give me another hour …

Reply:
`ApiResult` is designed to be serialized (a Result that can go over an API). It doesn't have all the amazing runtime methods of `Result`, but just stores the same info in a serialized form.

I think you had some issues with `ApiResult` before as well (which I understand). Maybe we can move it to another issue? I would like to get this PR in so I can build on top of it. And we can refine the use of `ApiResult`, the choice of JSON lib, etc. in follow-up PRs.

If the request is to use different names between the two packages, that I understand well and am happy to rename the symbols in `cosmwasm_vm::testing`, any suggestions?

Reply:
Feel free to merge as is. I'll cleanup the two difeerent implementations of `InitResult` shortly

Reply:
Thank you.